### PR TITLE
Amanda backup system

### PR DESCRIPTION
**Pull request by mstormi**
Initial patch to introduce the Amanda backup tool to openHABian.

Please bear with me if it doesn't work right away. It works as a standalone shell script based on my Raspian installation, but I don't have a proper openHABian testing environment, so there likely can be issues with user rights, file permissions etc.

Signed-off-by: Markus Storm <markus.storm@gmx.net> (github: mstormi)

### CODE REVIEW

**Comment by ThomDietrich** [comment]
Thanks! I didn't have a deeper look yet. One first suggestion would be to outsource the config files. Usage of ` /bin/cat<<EOF` clutters sourcecode and is hard to control. My preference would be a file in `include/` as with samba or nginx.

**Comment by mstormi** [comment]
Removed inline file creation

File: openhabian-setup.sh [hunk 1]
```diff
@@ -1151,6 +1151,167 @@ nginx_setup() {
...
+    if (whiptail --title "Create Amazon AWS based backup" --yes-button "Yes" --no-button "No" --yesno "Setup a backup mechanism based on Amazon Web Services. You can get 5 GB of S3 cloud storage for free on https://aws.amazon.com/. See also http://wiki.zmanda.com/index.php/How_To:Backup_to_Amazon_S3" 15 80) then
+        config=openhab-AWS
+        S3accesskey=$(whiptail --title "S3 access key" --inputbox "Enter the S3 access key you obtained at S3 setup time:" 10 60 3>&1 1>&2 2>&3)
+        S3secretkey=$(whiptail --title "S3 secret key" --inputbox "Enter the S3 secret key you obtained at S3 setup time:" 10 60 3>&1 1>&2 2>&3)
+        tapes=$(whiptail --title "Number of virtual storage containers in rotation" --inputbox "How many virtual containers will you setup inside the S3 bucket ?  Note that #container x container size will need to fit into your S3 bucket." 10 60 3>&1 1>&2 2>&3)
+        size=$(whiptail --title "Container size" --inputbox "How large do you want one virtual container to be ? Specify the size in megabytes." 10 60 3>&1 1>&2 2>&3)
+
+        create_backup_config ${config} ${backupuser} ${tapes} ${size} AWS ${S3accesskey} ${S3secretkey}
+    fi
+  fi
```
Comment (ThomDietrich) [inline openhabian-setup.sh]: Is there a realistic else action? If the function can only be executed interactively you can make that decision once at the top. (`echo "FAILED"; exit 1;`)

File: openhabian-setup.sh [hunk 2]
```diff
@@ -1151,6 +1151,167 @@ nginx_setup() {
...
+     introtext="${introtext}\nFor permanent storage such as USB or NAS mounted storage, as well as for cloud based storage, we will create ${tapes} virtual containers."
+  fi
+  if [ -n "$INTERACTIVE" ]; then
+      if ! (whiptail --title "Storage container creation" --yes-button "Continue" --no-button "Back" --yesno "$introtext" 15 80) then return 0; fi
+  fi
+  # create 'tapes'
+  counter=1
+  while [ ${counter} -le ${tapes} ]; do
+      if [ "${config}" = "openhab-dir" ]; then
+echo          mkdir -p ${storage}/slots/slot${counter}
```
Comment (ThomDietrich) [inline openhabian-setup.sh]: Typo?

File: includes/amanda.conf_template [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	compress client fast                # Compress in client side with less CPU (fast)
+}
+define application-tool app_amraw {   # how to dump the SD card's raw device /dev/mmcblk0
+        plugin "amraw"                # uses 'dd'
+}
+define dumptype amraw {
+        global
+        program "APPLICATION"
+        application "app_amraw"
+}
```
Comment (ThomDietrich) [inline includes/amanda.conf_template]: Rename file to .conf or add the following in the last line:

```
# vim: filetype=conf
```

Also: Are the comments supposed to be indented equally?

File: openhabian-setup.sh [hunk 4]
```diff
@@ -1151,6 +1151,167 @@ nginx_setup() {
...
+            password=$(whiptail --title "Authentication Setup" --passwordbox "Enter a password for $backupuser:" 15 80 3>&1 1>&2 2>&3)
+            secondpassword=$(whiptail --title "Authentication Setup" --passwordbox "Please confirm the password:" 15 80 3>&1 1>&2 2>&3)
+            if [ "$password" = "$secondpassword" ] && [ ! -z "$password" ]; then
+                matched=true
+            else
+                password=$(whiptail --title "Authentication Setup" --msgbox "Password mismatched or blank... Please try again!" 15 80 3>&1 1>&2 2>&3)
+            fi
+      done
+  fi
+  echo ${backupuser}:${password} | /usr/bin/passwd ${backupuser}
```
Comment (ThomDietrich) [inline openhabian-setup.sh]: Alternative:
```
chpasswd <<< "$backupuser:$password"
```

File: openhabian-setup.sh (hunk 2 above)
Comment (mstormi) [inline openhabian-setup.sh]: yes, testing leftover

File: includes/amanda.conf_template (hunk 3 above)
Comment (mstormi) [inline includes/amanda.conf_template]: done

File: openhabian-setup.sh (hunk 4 above)
Comment (mstormi) [inline openhabian-setup.sh]: if you prefer ....

File: openhabian-setup.sh (hunk 1 above)
Comment (mstormi) [inline openhabian-setup.sh]: Still not sure. Does not do harm, so leave it as-is

File: includes/amanda.conf_template (hunk 3 above)
Comment (rtvb) [inline includes/amanda.conf_template]: I use nano. Will syntax formatting work for the file type?

Comment (mstormi) [inline includes/amanda.conf_template]: Oh dear. Why don't you just download the file and try for yourself?
It's highlighting of a syntax that vim/nano don't (fully) understand, so it's of little use anyway.

Comment (ThomDietrich) [inline includes/amanda.conf_template]: @rtvb most editors react on the file ending. The vim modeline is a vim workaround but will not have any effect in nano.

**Comment by ThomDietrich** [comment]
@mstormi I'm traveling this week. I'll have a look over the next week (hopefully). @rtvb did you do any testing?

**Comment by ThomDietrich** [comment]
Hey @mstormi, finally got around to look into this. Thanks for your contribution!! Feel free to add improvements as you go. Very much appreciated ;)

**Pull request merged**